ControlSettings/ControlSettings.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include "ControlSettings.h"
+
+const char* PLUGIN_IARM_BUS_NAME = "Thunder_Plugins";
+
+#define registerMethod(...) Register(__VA_ARGS__);GetHandler(2)->Register<JsonObject, JsonObject>(__VA_ARGS__)
+
+namespace WPEFramework {
+namespace Plugin {
+
+    SERVICE_REGISTRATION(ControlSettings,1, 0);
+
+    ControlSettings::ControlSettings()
+               : _skipURL(0)
+    {
+        LOGINFO("Entry\n");
+
+	InitializeIARM();
+	CreateHandler({ 2 });
+
+        //Common API Registration
+
+        LOGINFO("Exit \n");
+    }
+
+    ControlSettings::~ControlSettings()
+    {
+        LOGINFO();
+    }
+
+    const std::string ControlSettings::Initialize(PluginHost::IShell* service)
+    {
+	LOGINFO("Entry\n");
+        
+	ASSERT(service != nullptr);
+        _skipURL = static_cast<uint8_t>(service->WebPrefix().length());
+
+	DEVICE_TYPE::getInstance()->Initialize();
+
+	LOGINFO("Exit\n");
+        return (service != nullptr ? _T("") : _T("No service."));
+    }
+
+    void ControlSettings::Deinitialize(PluginHost::IShell* service)
+    {
+        LOGINFO();
+
+	DEVICE_TYPE::getInstance()->Deinitialize();
+    }
+
+    void ControlSettings::InitializeIARM()
+    {
+#if !defined (HDMIIN_4K_ZOOM)
```

Comment:
why IARM init is linked with HDMIIN_4K_ZOOM

Reply:
I have just followed the current tvsettings-plugins.
This flag(HDMIIN_4K_ZOOM) introduced as part of RDK-34823.